modules/currency-l0/src/main/scala/org/tessellation/currency/l0/Main.scala

```diff
@@ -144,7 +147,11 @@ object Main
             NodeState.Initial,
             NodeState.LoadingGenesis,
             NodeState.GenesisReady
-          )(programs.genesis.accept(m.genesisPath)) >> gossipDaemon.startAsInitialValidator >>
+          ) {
+            dataApplication.traverse { da =>
+              da.serializeState(da.genesis)
+            }.flatMap(programs.genesis.accept(m.genesisPath, _))
```

Comment:
`da.serializeState(da.genesis)` seems to operate on the wrong abstraction level. You could hide the details in something like: `da.serializedGenesis` so `Main` doesn't need to know the details.

Reply:
:+1: